src/System.Private.CoreLib/shared/System/AppDomain.cs

```diff
@@ -24,18 +25,18 @@ public sealed partial class AppDomain : MarshalByRefObject
     {
         private static readonly AppDomain s_domain = new AppDomain();
         private readonly object _forLock = new object();
-        private IPrincipal _defaultPrincipal;
+        private IPrincipal? _defaultPrincipal;
         private PrincipalPolicy _principalPolicy = PrincipalPolicy.NoPrincipal;
-        private Func<IPrincipal> s_getWindowsPrincipal;
-        private Func<IPrincipal> s_getUnauthenticatedPrincipal;
+        private Func<IPrincipal>? s_getWindowsPrincipal;
+        private Func<IPrincipal>? s_getUnauthenticatedPrincipal;
 
         private AppDomain() { }
 
         public static AppDomain CurrentDomain => s_domain;
 
-        public string BaseDirectory => AppContext.BaseDirectory;
+        public string? BaseDirectory => AppContext.BaseDirectory;
```

Comment:
Interesting, I didn't know `AppDomain/AppContext.BaseDirectory` could return null, good to know.